### DISCUSSION TITLE
Improved 2D lighting

### DISCUSSION THREAD

**Discussion by PoisonousGame** [godotengine/godot-proposals#5384] [2D]
### Describe the project you are working on

I'm working on a game with a day/night cycle.

### Describe the problem or limitation you are having in your project

I'm working on a game with a day/night cycle. If you use the built-in lighting system, multiple light sources are exposed when stacked on top of each other.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I'm working on a game with a day/night cycle. If you use the built-in lighting system, multiple light sources are exposed when stacked on top of each other. This is not the effect I want. The effect I want looks like this:https://www.youtube.com/watch?v=XaJ58TxKTTc&t=199s
I don't think most games would want exposure when multiple light sources are stacked on top of each other, godot should have this built in to prevent exposure.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I don't know how to code in C++

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can now only be achieved through shaders

### Is there a reason why this should be core and not an add-on in the asset library?

Games that require the use of light do not want to expose, should be built-in

**Comment by SilencedPerson** [comment]
People have been asking for this for years and it's unlikely to happen as light is very low level and there simply isn't any capable/available contributor to do this and none of the main guys have any interest in it as it can be done with shaders. You will have to do this with shaders too, sadly.

**Comment by Zireael07** [comment]
Should've been a discussion, as I see no code proposal here. Also, the title is very generic... 'do not stack multiple 2d light sources' would've probably been a better one

**Comment by SilencedPerson** [comment]
In my opinion it's duplicate of #4282 or #2710

Edit: here is solution to your problem using shaders.
[Link to Godot Shaders](https://godotshaders.com/shader/masking-shader/)
[Link to the Video Tutorial on Youtube](https://www.youtube.com/watch?v=Fb0MM4u5lWs)
